src/Logs.tsx

```diff
@@ -0,0 +1,264 @@
+import React, { useState, useContext, useEffect, ReactElement } from 'react';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
+import moment from 'moment';
+import { Button, Dropdown, Header, Table, Icon } from 'semantic-ui-react';
+import { NumericOnlyInput } from './components';
+import { LogStatsContext, ClickedDayContext } from './Context';
+import { addExerciseToDatabase, retrieveExerciseDatabase, addLogToDatabase, retrieveLogDatabase, removeLogFromDatabase } from './Database';
+import './index.css';
+
+function Logs(): ReactElement {
+  const clickedDay = useContext(ClickedDayContext);
+  const [currentTab, setcurrentTab] = useState(<div />);
+
+  function handleClickDay(date: Date) {
+    clickedDay.setDate(moment(date));
+    setcurrentTab(<div />);
+  }
+
+  function changeDayBackgroundIfHasLog(date: Date, view: string): string | null {
+    const dayHasLog = Object.prototype.hasOwnProperty.call(retrieveLogDatabase(), moment(date).format('DD MM YYYY'));
+    return view === 'month' && dayHasLog ? 'dayWithLog' : null;
+  }
+
+  function changeDayBackgroundIfFuture(date: string, view: string): string | null {
+    const isInFuture = moment().isBefore(date);
+    return view === 'month' && isInFuture ? 'dayInFuture' : null;
+  }
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
+      <Calendar onClickDay={handleClickDay} tileClassName={({ date, view }) => [changeDayBackgroundIfHasLog(date, view), changeDayBackgroundIfFuture(date, view)]} />
+      <LogScreenButtons currentScreen={setcurrentTab} />
+      {currentTab}
+    </div>
+  );
+}
+
+function LogScreenButtons(props: { currentScreen: React.Dispatch<React.SetStateAction<JSX.Element>> }): ReactElement {
+  const { currentScreen } = props;
```

Comment:
Usually the way you do this is you pass an onClick function to the LogScreenButtons component, and the component calls that function with the required arguments, instead of passing dispatches as a prop.

Reply:
fixed